src/synthesizer/python/mod.rs

```diff
@@ -0,0 +1,645 @@
+        ResourceIr::ImportValue(name) => {
+            output.text(format!("cdk.Fn.importValue('{}')", name.escape_debug()))
+        }
+        ResourceIr::Join(sep, list) => {
+            let items = output.indent_with_options(IndentOptions {
+                indent: INDENT,
+                leading: Some("[".into()),
+                trailing: Some(format!("].join('{sep}')", sep = sep.escape_debug()).into()),
+                trailing_newline: false,
+            });
+            for item in list {
+                emit_resource_ir(context, &items, item, Some(",\n"));
+            }
+        }
+        ResourceIr::Map(name, tlk, slk) => {
+            output.text(format!("{}[", camel_case(name)));
+            emit_resource_ir(context, output, tlk, None);
+            output.text("][");
+            emit_resource_ir(context, output, slk, None);
+            output.text("]")
+        }
+        ResourceIr::Select(idx, list) => match list.as_ref() {
+            ResourceIr::Array(_, array) => {
+                if *idx <= array.len() {
+                    emit_resource_ir(context, output, &array[*idx], None)
+                } else {
+                    output.text("None")
+                }
+            }
+            other => {
+                output.text("cdk.Fn.select(");
+                output.text(idx.to_string());
+                output.text(", ");
+                emit_resource_ir(context, output, other, None);
+                output.text(")")
+            }
+        },
+        ResourceIr::Split(sep, str) => match str.as_ref() {
+            ResourceIr::String(str) => {
+                output.text(format!("'{str}'", str = str.escape_debug()));
+                output.text(format!(".split('{sep}')", sep = sep.escape_debug()))
+            }
+            other => {
+                output.text(format!("cdk.Fn.split('{sep}', ", sep = sep.escape_debug()));
+                emit_resource_ir(context, output, other, None);
+                output.text(")")
+            }
+        },
+        ResourceIr::Sub(parts) => {
+            output.text("'");
+            for part in parts {
+                match part {
+                    ResourceIr::String(lit) => output.text(lit.clone()),
+                    other => {
+                        output.text("{");
+                        emit_resource_ir(context, output, other, None);
+                        output.text("}");
+                    }
+                }
+            }
+            output.text("'")
+        }
+
+        // References
+        ResourceIr::Ref(reference) => output.text(reference.to_python()),
+    }
+    if let Some(trailer) = trailer {
+        output.text(trailer.to_owned())
+    }
+}
+
+fn append_references(output: &CodeBuffer, reference: &ResourceInstruction) {
+    for dep in &reference.references {
+        output.line(format!("if ({dep} is None): raise Exception(\"A combination of conditions caused '{dep}' to be None. Fixit.\")", dep=camel_case(dep)));
```

Comment:
This error message feels a little clunky. Can we make it more useful to the user?

Reply:
Even if we can't provide more information the message, as it stands, seems pretty idiomatic to Rust, not to the target language so we should edit it to reflect python conventions instead of Rust. Also, `Fixit` probably isn't the wording we want there.

Reply:
Granted, I don't actually use Python so maybe I'm wrong here. I just saw that same-ish style in TypeScript as well.

Reply:
for context, there was a conversation offline and we agreed that this actually serves 0 practical purpose so we are removing it.